### ISSUE TITLE
readdir does not return filepaths exceeding MAX_PATH on windows

### ISSUE THREAD

**Issue by jkommer** [mpv-player/mpv#11539]
I'm using the autoload.lua script to automatically create playlists for series but noticed that at times I'm either missing files or no playlist gets created. After some playing around this seems to be due to an issue with paths exceeding MAX_PATH length as `utils.readdir` does not seem to return any files exceeding this length.

### Important Information

- mpv version: 0.35.0.348-g0f13c38e
- Windows Version: windows 10, 22H2 (x64)
- Source of the mpv binary: shinchiro (x64)
- Windows long path support is enabled

### Reproduction steps

- Create a directory approaching the `MAX_PATH` (e.g. `F:\a91bNp9Rc5XFJkz2owSSd30xZPd3pP6TKpekjGJWJolvGzCzVS9npo3CentKR2VLrc19pJynXfi2J7Alb2qsU1zNBKnAIWIIDF9LeBMfwyLhF7VOD8KwGLiBGaHInhlEwlOLbvZWYRlFxmDFnlXPjFqftp0N4zVyaXd1yxJ5iOhavfolotoB5K0SlF1NcU224wfrAVI7qdhUGAI8s1OMHqJwbMtlpwydfPE5pSK2ckSp0NuHOlIb`
- Add three video files; `123.kmv`, `12345.mkv` and `123456789012345678.mkv`
- Call `utils.readdir` against this directory through the following script

```
local msg = require 'mp.msg'
local utils = require 'mp.utils'

function find_and_add_entries()
    local files = utils.readdir("F:\\a91bNp9Rc5XFJkz2owSSd30xZPd3pP6TKpekjGJWJolvGzCzVS9npo3CentKR2VLrc19pJynXfi2J7Alb2qsU1zNBKnAIWIIDF9LeBMfwyLhF7VOD8KwGLiBGaHInhlEwlOLbvZWYRlFxmDFnlXPjFqftp0N4zVyaXd1yxJ5iOhavfolotoB5K0SlF1NcU224wfrAVI7qdhUGAI8s1OMHqJwbMtlpwydfPE5pSK2ckSp0NuHOlIb", "files")

    for i = 1, #files do
        msg.error("readdir returned file: " .. files[i])
    end
end

mp.register_event("start-file", find_and_add_entries)

```

### Expected behavior

All three files to be returned.

### Actual behavior

Only those files underneath the MAX_PATH length are returned. 

### Log file

```
[   0.019][v][cplayer] Running hook: auto_profiles/on_load
[   0.019][e][autoload] readdir returned file: 123.mkv 
[   0.019][e][autoload] readdir returned file: 12345.mkv 
[   0.019][v][ifo_dvdnav] Opening F:\a91bNp9Rc5XFJkz2owSSd30xZPd3pP6TKpekjGJWJolvGzCzVS9npo3CentKR2VLrc19pJynXfi2J7Alb2qsU1zNBKnAIWIIDF9LeBMfwyLhF7VOD8KwGLiBGaHInhlEwlOLbvZWYRlFxmDFnlXPjFqftp0N4zVyaXd1yxJ5iOhavfolotoB5K0SlF1NcU224wfrAVI7qdhUGAI8s1OMHqJwbMtlpwydfPE5pSK2ckSp0NuHOlIb\123.mkv
```

**Comment by Dudemanguy** [comment]
Might have been fixed by https://github.com/mpv-player/mpv/commit/9565675488c51133e85627e5e610bf631eca6289? Can anyone confirm?

**Comment by jkommer** [comment]
This does indeed seem to be fixed in mpv v0.37.0-135-gabc2a748 - all three of the expected files are now returned on Windows

```
         x: readdir returned file: 123.mkv
         x: readdir returned file: 12345.mkv
         x: readdir returned file: 123456789012345678.mkv
```
